packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -1699,6 +1707,86 @@ def __parse_rpc_error(error: str) -> RPCResponseStatus:
             return RPCResponseStatus.ALREADY_KNOWN
         return RPCResponseStatus.UNCLASSIFIED_ERROR
 
+    def _acn_request_from_pending(
+        self, performative: TendermintMessage.Performative
+    ) -> Generator:
+        """Perform an ACN request to each one of the agents which have not sent a response yet."""
+        not_responded_yet = {
+            address
+            for address, deliverable in cast(
+                SharedState, self.context.state
+            ).address_to_acn_deliverable.items()
+            if deliverable is None
+        }
+
+        if len(not_responded_yet) == 0:
+            return
+
+        self.context.logger.debug(f"Need ACN response from {not_responded_yet}.")
+        for address in not_responded_yet:
+            self.context.logger.debug(f"Sending ACN request to {address}.")
+            dialogues = cast(TendermintDialogues, self.context.tendermint_dialogues)
+            message, _ = dialogues.create(
+                counterparty=address, performative=performative
+            )
+            message = cast(TendermintMessage, message)
+            context = EnvelopeContext(connection_id=P2P_LIBP2P_CLIENT_PUBLIC_ID)
+            self.context.outbox.put_message(message=message, context=context)
+
+        # we wait for the `address_to_acn_deliverable` to be populated with the responses (done by the tm handler)
+        yield from self.sleep(self.params.sleep_time)
```

Comment:
maybe we can replace this with a condition check, with a timeout?

Reply:
Sure, but this would require that we replace the retry logic:

https://github.com/valory-xyz/open-autonomy/blob/0e3cf01ec9a882f401868049fd419e807f577f71/packages/valory/skills/abstract_round_abci/behaviour_utils.py#L1756-L1766

So I would suggest that we address this on a separate PR at some point.